src/Core/Domain/Category/ValueObject/CategoryId.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Category\ValueObject;
+
+/**
+ * Class CategoryId
+ */
+class CategoryId
+{
+    /**
+     * @var int
+     */
+    private $categoryId;
+
+    /**
+     * @param int $categoryId
+     */
+    public function __construct($categoryId)
+    {
+        $this->setCategoryId($categoryId);
+    }
+
+    /**
+     * @return int
+     */
+    public function getValue()
+    {
+        return $this->categoryId;
+    }
+
+    /**
+     * @param CategoryId $categoryId
+     *
+     * @return bool
+     */
+    public function isEqual(CategoryId $categoryId)
+    {
+        return $this->getValue() === $categoryId->getValue();
+    }
+
+    /**
+     * @param int $categoryId
+     */
+    private function setCategoryId($categoryId)
```

Comment:
What is the need for a private setter ?

Reply:
If you want to change the id, you need to generate a new class.